src/comm.c

```diff
@@ -2229,6 +2230,11 @@ get_message (char *buff, size_t *bufflength)
                     continue;
                 }
 
+                /* Process telnet negotiations up to the point where the
+                 * next input is ready, if possible.
+                 */
+                telnet_neg(ip);
```

Comment:
Note: This is the key difference to the previous version of the patch that lead to slow processing of player inputs.